### ISSUE TITLE
Can't use Google in iFrame

### ISSUE THREAD

**Issue by audition** [audition/guava#4]
We can't put Google inside of an iFrame. Google sends a `"X-Frame-Options: SAMEORIGIN"` response header. This means the page can only be displayed in a frame on the same origin as the page itself. Typically this a feature to prevent click jacking, but a side effect is no iFrames.

An alternative would be to use Google's custom search feature (listed here: https://developers.google.com/custom-search/)

Although it does cost $$ - and may not be something we want to invest our time into. Another option would be to use Bing search (free API): http://www.bing.com/developers/s/APIBasics.html

Would love to know everyone's thoughts.
@ambientwanderer @vixdug @0ptimusRhyme @NicholasZhang @sdrupert

**Comment by vixdug** [comment]
Hmmm. I am no Bing fan, that is for sure. What is the license cost for Google custom search? How much bigger of a project would it be to implement it? I mean I like the idea of it, minus the cost.

**Comment by audition** [comment]
[$100 a year for under 20k searches.](https://support.google.com/customsearch/answer/72334?hl=en) Looking into this further, it looks like there's a [free version](https://support.google.com/adsense/answer/9879?ctx=as2&hl=en&rd=2&ref_topic=1705820), but it's more with targeted ads, and monetization of the searches. Not sure if that's a path we want to go down.

If there are no objections, I will try and use the free CSE version, and see if it's something we want to use. In the interim, we should think and explore about this more.

**Comment by ambientwanderer** [comment]
@audition I did try out bing's search and it isn't terrible for results when I tried out a few things so it would make for a good backup plan if the free version doesn't work out as well.

**Comment by 0ptimusRhyme** [comment]
I'm going to investigate any alternatives so that we can use Google. I have
a feeling Bing just won't deliver and let's be honest, it looks worse for
those using it.

On Thu, Mar 24, 2016 at 1:40 PM, Travis Currie <notifications@github.com>
wrote:

> @audition <https://github.com/audition> I did try out bing's search and
> it isn't terrible for results when I tried out a few things so it would
> make for a good backup plan if the free version doesn't work out as well.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/audition/guava/issues/4#issuecomment-201010743>
>



-- 
Alex
Customer Success Guru
_shopify.com_

**Comment by audition** [comment]
Here's an example of what we could do with Google's free version of CSE: https://thedohm.myshopify.com/

**Comment by 0ptimusRhyme** [comment]
Not pretty:




On Thu, Mar 24, 2016 at 1:46 PM, Elliot Dohm <notifications@github.com>
wrote:

> Here's an example of what we could do with Google's free version of CSE:
> https://thedohm.myshopify.com/
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/audition/guava/issues/4#issuecomment-201013581>
>



-- 
Alex
Customer Success Guru
_shopify.com_

**Comment by ambientwanderer** [comment]
I just ran a basic search and it failed. Was looking for shopify payments and this is what I got 
![screencap](https://monosnap.com/file/HCQUfFFIT1TMtt3ImEJwvQgm6FCLZb)

**Comment by vixdug** [comment]
^ not liking that at all Travis hah

**Comment by ambientwanderer** [comment]
What about this here it works a little better and still free http://ambientwanderlust2.myshopify.com/

**Comment by audition** [comment]
Still not a huge fan of the display :( @ambientwanderer what search terms did you use to define Google's CSE?

**Comment by ambientwanderer** [comment]
I told it to just look at the forums, docs and shopify.ca to get the results @audition .If that is really all we are pulling from wouldn't it make sense to not use google and just use those plus github and zendesk to create this search engine avoid google altogether.

**Comment by audition** [comment]
This is a good point - though there are other resources Guru's may be using to search, that are not defined by docs or forums. I say we continue with project development, and reevaluate incorporating Google, or another search engine later. What do you think? @ambientwanderer

**Comment by ambientwanderer** [comment]
I agree, lets just put google on hold, if we can prove that this project will really help support gurus and other support teams then maybe shopify would be willing to pay for the proper google search. I think we should check in with a few gurus make sure there isn't anything else they are using we could use. The only other thing I could think of maybe is stack overflow for certain issues. @audition
